src/mapped-types.ts

```diff
@@ -412,12 +412,18 @@ export type PromiseType<T extends Promise<any>> = T extends Promise<infer U>
  *   };
  *   type ReadonlyNestedProps = DeepReadonly<NestedProps>;
  */
-export type DeepReadonly<T> = T extends (...args: any[]) => any
+export type DeepReadonly<T> = T extends
+  | ((...args: any[]) => any)
+  | string
+  | number
+  | boolean
+  | null
+  | undefined
```

Comment:
What is the rationale behind null and undefined here? If there is something please add test cases to cover it.

Reply:
Completeness, doesn't look necessary (added a test) so I removed it.